dotCMS/src/main/java/com/dotcms/vanityurl/handler/DefaultVanityUrlHandler.java

```diff
@@ -27,6 +31,19 @@
     private static final int LIMIT = 2;
 
     @Override
+    public VanityUrlResult handle(final String uri,
+            final HttpServletResponse response, final Host host,
+            final long languageId, final User user) throws IOException {
+
+        CachedVanityUrl vanityUrl = APILocator.getVanityUrlAPI()
+                .getLiveCachedVanityUrl(("/".equals(uri) ? CMS_HOME_PAGE
+                                : uri.endsWith("/") ? uri.substring(0, uri.length() - 1) : uri), host,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested ternary operation into an independent statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3358)